xray/commands/audit/scan.go

```diff
@@ -0,0 +1,270 @@
+	return func(filePath string) parallel.TaskFunc {
+		return func(threadId int) (err error) {
+
+			logMsgPrefix := clientutils.GetLogMsgPrefix(threadId, false)
+			fileInfo, e := os.Lstat(filePath)
+			if errorutils.CheckError(e) != nil {
+				return
+			}
+			log.Info(logMsgPrefix+"Indexing file:", fileInfo.Name())
+			graph, err := scanCmd.IndexFile(filePath)
+			if err != nil {
+				return err
+			}
+			// Add a new task to the seconde prodicer/consumer
+			// which will send the indexed binary to Xray and then will store the given result.
+			taskFunc := func(threadId int) (err error) {
+				scanResults, err := scanCmd.GetXrScanGraphResults(graph)
+				if err != nil {
+					return err
+				}
+				resultsArr[threadId] = append(resultsArr[threadId], scanResults)
+				return
+			}
+
+			indexedFileProducer.AddTaskWithError(taskFunc, errorsQueue.AddError)
+			return
+		}
+	}
+}
+
+func getAddTaskToProducerFunc(producer parallel.Runner, errorsQueue *clientutils.ErrorsQueue, fileHandlerFunc FileContext) indexFileHandlerFunc {
+	return func(filePath string) {
+		taskFunc := fileHandlerFunc(filePath)
+		producer.AddTaskWithError(taskFunc, errorsQueue.AddError)
+	}
+}
+
+func (scanCmd *XrBinariesScanCommand) performScanTasks(fileConsumer parallel.Runner, indexedFileConsumer parallel.Runner, resultsArr [][]*services.ScanResponse) {
+
+	go func() {
+		// Blocking until consuming is finished.
+		fileConsumer.Run()
+		// After all files has been indexed, The seconde producer notifies that no more tasks will be produced.
+		indexedFileConsumer.Done()
+	}()
+	// Blocking until consuming is finished.
+	indexedFileConsumer.Run()
+	// Prints the results
+	for _, arr := range resultsArr {
+		for _, res := range arr {
+			printTable(res)
+		}
+	}
+	return
+}
+
+func collectFilesForIndexing(fileData spec.File, dataHandlerFunc indexFileHandlerFunc) error {
+
+	fileData.Pattern = (clientutils.ReplaceTildeWithUserHome(fileData.Pattern))
+	// Save parentheses index in pattern, witch have corresponding placeholder.
+	patternType := clientutils.WildCardPattern
+	if regex, _ := fileData.IsRegexp(false); regex {
+		patternType = clientutils.RegExp
+	}
+	if ant, _ := fileData.IsAnt(false); ant {
+		patternType = clientutils.AntPattern
+	}
+
+	rootPath, err := fspatterns.GetRootPath(fileData.Pattern, fileData.Target, patternType, false)
+	if err != nil {
+		return err
+	}
+
+	isDir, err := fileutils.IsDirExists(rootPath, false)
+	if err != nil {
+		return err
+	}
+
+	// If the path is a single file, index it and return
+	if !isDir {
+		dataHandlerFunc(rootPath)
+		return nil
+	}
+	fileData.Pattern = clientutils.PrepareLocalPathForUpload(fileData.Pattern, patternType)
+	err = collectPatternMatchingFiles(fileData, rootPath, dataHandlerFunc)
+	return err
+}
+
+func collectPatternMatchingFiles(fileData spec.File, rootPath string, dataHandlerFunc indexFileHandlerFunc) error {
+	fileParams, err := fileData.ToArtifactoryCommonParams()
```

Comment:
ToArtifactoryCommonParams --> ToCommonParams

The ```utils.ArtifactoryCommonParams``` struct should also be renamed to ```utils.CommonParams```.